### PR TITLE
Support for Docker and Podman build on mac silicon (ARM)

### PR DESCRIPTION
**Pull request by zubcevic**
This makes it possible to run the container builds on MacOS with silicon chips as well as other platforms that have either podman or docker installed.

To use:
unchanged: ./mvnw clean install -PbuildDocker 

or if you prefer podman:
./mvnw clean install -PbuildDocker -Dcontainer.executable=podman

### CODE REVIEW

**Comment by zubcevic** [comment]
#180

File: pom.xml [hunk 1]
```diff
@@ -122,35 +122,36 @@
...
+                                            <argument>DOCKERIZE_VERSION=${docker.image.dockerize.version}</argument>
+                                            <argument>-t</argument>
+                                            <argument>${docker.image.prefix}/${project.artifactId}</argument>
+                                            <argument>${project.build.directory}</argument>
+                                        </arguments>
+                                    </configuration>
                                 </execution>
                             </executions>
-                            <configuration>
-                                <imageName>${docker.image.prefix}/${project.artifactId}</imageName>
```
Comment (arey) [inline pom.xml:137]: What I see is that you've replaced the use of the spotify plugin with a call to the docker CLI (or podman). 
I have the impression that you've lost some functionality, such as the ability to push the image into Docker Hub.
Am I mistaken?

Comment (zubcevic) [reply pom.xml:137]: I didn’t see documentation about pushing containers to Docker Hub or other repositories. I did check for a github workflow that might do this. The spotify plugin looks retired. As the container build with this plugin requires podman or docker, those tools can easily be used for pushing. Podman compose can also be used to start the application if build with podman.

Comment (zubcevic) [reply pom.xml:137]: https://github.com/spotify/docker-maven-plugin

Comment (arey) [reply pom.xml:137]: I don't think I wrote it in the `readme.md` but I was using this kind of command line to publish all docker images to docker hub:
```
mvn install -PbuildDocker -DpushImageTag -DdockerImageTags=2.1.2,java11
```
I know that both Spotify projects [docker-maven-plugin](https://github.com/spotify/docker-maven-plugin) and [dockerfile-maven](https://github.com/spotify/dockerfile-maven) are stoped. Maybe it's time to remove it. I could plan to use the Spring Boot buildpacks support.

Comment (arey) [reply pom.xml:137]: Could you please add to the readme a comment to switch from docker to podman:
```
or if you prefer podman:
./mvnw clean install -PbuildDocker -Dcontainer.executable=podman
```

Comment (zubcevic) [reply pom.xml:137]: documentation has been added

**Comment by arey** [comment]
Thank you for your contribution René

**Comment by arey** [comment]
Hi @zubcevic 

I recovered these changes. Building docker image on my Intel Macbook and Docker Machine is working.
But I've got the follwing error while starting the `discovery-server`
```
arey:~$docker-compose up                 
grafana-server is up-to-date
prometheus-server is up-to-date
config-server is up-to-date
tracing-server is up-to-date
Starting discovery-server ... error

ERROR: for discovery-server  Cannot start service discovery-server: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "./dockerize": stat ./dockerize: no such file or directory: unknown
```
Do you have any idea?

**Comment by arey** [comment]
Fixed by https://github.com/spring-petclinic/spring-petclinic-microservices/pull/246

**Pull request merged**